File: CS/12_Tables/AdjustColumnByTextWidth.cs
using System;
using System.Windows.Forms;
using Spire.Presentation;

namespace AdjustColumnByTextWidth
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load the file from disk.
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\Template_Ppt_1.pptx");

            //Get the table from the first slide of the sample document.
            ISlide slide = presentation.Slides[0];
            ITable table = slide.Shapes[0] as ITable;

            //Adjust the first column width of table by text width.
            table.ColumnsList[0].AdjustColumnByTextWidth();



            //Save to file.
            String result = "output.pptx";
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the PowerPoint file.
            PptDocumentViewer(result);
        }

        private void PptDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}

File: VB/12_Tables/AdjustColumnByTextWidth.vb
Imports Spire.Presentation

Namespace AdjustColumnByTextWidth
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load the file from disk.
			presentation.LoadFromFile("..\..\..\..\..\..\Data\Template_Ppt_1.pptx")

			'Get the table from the first slide of the sample document.
			Dim slide As ISlide = presentation.Slides(0)
			Dim table As ITable = TryCast(slide.Shapes(0), ITable)

			'Adjust the first column width of table by text width.
			table.ColumnsList(0).AdjustColumnByTextWidth()



			'Save to file.
			Dim result As String = "output.pptx"
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the PowerPoint file.
			PptDocumentViewer(result)
		End Sub

		Private Sub PptDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace